lib/python/rose/apps/rose_bunch.py

```diff
@@ -0,0 +1,458 @@
+# -*- coding: utf-8 -*-
+# ----------------------------------------------------------------------------
+# (C) British Crown Copyright 2012-5 Met Office.
+#
+# This file is part of Rose, a framework for meteorological suites.
+#
+# Rose is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Rose is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Rose. If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+"""Builtin application: rose_bunch: run multiple commands in parallel."""
+
+
+import os
+import shlex
+import sqlite3
+from multiprocessing import Pool
+from time import sleep
+
+from rose.env import env_var_process
+from rose.app_run import (
+    BuiltinApp,
+    ConfigValueError,
+    CompulsoryConfigValueError)
+from rose.popen import RosePopener
+import rose.job_runner
+from rose.reporter import Event
+
+
+class AbortEvent(Event):
+
+    """An event raised when a task failure will stop further task running"""
+
+    LEVEL = Event.V
+    KIND = Event.KIND_OUT
+
+    def __str__(self):
+        return "Closing task pool, no further commands will be run."
+
+
+class FailEvent(Event):
+
+    """An event used to report failure of a task in the pool."""
+
+    KIND = Event.KIND_ERR
+
+    def __str__(self):
+        name, rc = self.args
+        return "Command %s failed with return code %s" % (name, rc)
+
+class InconsistentArglength(Exception):
+
+    """An event raised when arg lengths are not consistent"""
+
+    ERROR_MSG = "Inconsistent arg lengths"
+
+    def __str__(self):
+
+        return  self.ERROR_MSG
+
+class LaunchEvent(Event):
+
+    """An event raised when adding a command to the task pool."""
+
+    LEVEL = Event.V
+    KIND = Event.KIND_OUT
+
+    def __str__(self):
+        name, cmd = self.args
+        return "Adding command %s to pool: %s" % (name, cmd)
+
+
+class SucceededEvent(Event):
+
+    """An event used to report success of a task in the pool."""
+
+    LEVEL = Event.V
+    KIND = Event.KIND_OUT
+
+    def __str__(self):
+        name, rc = self.args
+        return "Command %s completed successfully # rc = %s" % (name, rc)
```

Comment:
I think a simpler `[ OK ] command ...` will be sufficient, `rose_ana` style. (If I remember correctly, `rose.reporter` allows you to specify an alternate `prefix` when reporting an event.)